[Reference Only] Implement Abyssea Systems

claywar opened the pull request:
### **NOTE: PR is being broken down into smaller chunks for review, and is for reference only!**  

Major thanks to MrSent and TheWhaler for providing a foundation to work with for all of this.  There's still a long way to go before this will be completed, but this PR will encompass getting the Abyssea system in place (no guarantees on mob accuracy!)

**Summary of Changes**
* Converts first two Abyssea quests and Visions of Abyssea Main Quests to Interaction Framework
* Adds support for Traverser stone accrual based on epoch after flagging "The Truth Beckons"
* Adds support for Sturdy Pyxis
* Adds support for Abyssea Lights
* Re-table mob light info to be zone-specific
* Adds global and NPC scripts for Veridical Confluxes.  This data is stored in the char_unlocks table, and accessed through add/getTeleport bindings.
* Adds support for Abyssea Atmas, including Group 2 (needs data added to them), and now requires lunar abyssite keyitems to add an Atma.
* Adds Searing Ward tethering based on visitant status; Initial 5 minutes is handled by the same effect, with no effect icon visible.
* Update visitant status effect to display proper countdown from 5 minutes and below, and eject the player to the appropriate zone should it expire
* Fixes Abyssea teleporter parameters based on flagging or completing the zone boss quest.
* Wire up other Abyssea teleport NPCs, and move code to global
* Refactors Traverser Stone NPCs to support giving multiple stones at once, show time remaining if out of stones
* Moves Traverser Stone NPCs to global, and wires up other zones
* Refactors and implements all functions for Conflux Surveyor, and moved to abyssea global
* Disables dedication effect inside of Abyssea zones
* Add Dominion NPC (Including Augmented items, but no Trophy support)
* Adds support for Dominion Ops Quests (Grauberg implemented)

**TODO:**
* Add support to reinfuse previous atma set, and see recent
* Add Atma Fabricant
* Add remaining NPCs to appropriate Abyssea zones
* Reward XP bonuses based on same-family kills and gold lights
* Implement remaining Dominion Tacticians and move to global
* Add Bastion Prefect NPC
* Add seal rewards and influence bonuses for Dominion Ops Quests
* Add Tracker NPC (Ex: High Wind (A))
* (Must Fix) Implement one-hour lockout if time expired (Not getting visitant does not cause a lockout on retail!)
* (Must Fix) Disable certain NPC interactions behind the searing ward if player does not have visitant status
* Capture time expiration with visitant status active to ensure messages are correct
* Split and organize abyssea globals to a sub-directory

<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

claywar committed:
[core] Add support for traverser stone calculations and migration script

claywar committed:
[interaction] Convert first two Abyssea quests to interaction framework

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
[interaction] Remove The Truth Beckons references in zone.lua files

claywar committed:
[core] Add additional lua bindings, and add deathType to CBattleEntity

Co-authored-by: Omnione <mrsentientone@gmail.com>
Co-authored-by: TheWhaler <jmwv18@gmail.com>

Revert restoreFromChest binding trickery

claywar committed:
[lua] Pull in Abyssea Mixins

Co-authored-by: Omnione <mrsentientone@gmail.com>
Co-authored-by: TheWhaler <jmwv18@gmail.com>

claywar committed:
[lua] Pull and begin refactoring Sturdy Pyxis Global

Co-authored-by: Omnione <mrsentientone@gmail.com>
Co-authored-by: TheWhaler <jmwv18@gmail.com>

claywar committed:
[lua] Abyssea GM Commands

Co-authored-by: Omnione <mrsentientone@gmail.com>
Co-authored-by: TheWhaler <jmwv18@gmail.com>

claywar committed:
[lua] Pull and begin refactoring Abyssea global

Co-authored-by: Omnione <mrsentientone@gmail.com>
Co-authored-by: TheWhaler <jmwv18@gmail.com>

claywar committed:
[lua] Add Abyssea Lights

Co-authored-by: Omnione <mrsentientone@gmail.com>
Co-authored-by: TheWhaler <jmwv18@gmail.com>

claywar committed:
[lua] Add Veridical Conflux NPC scripts to Abyssea zones

claywar committed:
[lua] Veridical Conflux Global

claywar committed:
[lua] Add Sturdy Pyxis NPC scripts to Abyssea zones

claywar committed:
[lua] Import additional globals and NPC scripts

Co-authored-by: Omnione <mrsentientone@gmail.com>
Co-authored-by: TheWhaler <jmwv18@gmail.com>

claywar committed:
[lua] Refactor Atma global to use single effect with sub IDs

claywar committed:
[lua] Add all Atma GM command, fix Atma global issues, add support for Group 2 Atma


Add Lunar Abyssite to Abyssea Atma Slot count function


Correct KI syntax

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
[lua] Refactor Cruor Prospector, add to Konschtat, La Theine, and Tahrongi

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Update Abyssea Settings, remove old global from branch


Remove settings references from globals

claywar committed:
[lua] Sort abyssea light data by zone, convert to local


Remove duplicate table entry

claywar committed:
[lua] Fix Veridical Conflux cost calculations, and prevent tabled data from being modified

claywar committed:
[lua] Add additional NPC scripts to Abyssea - La Theine


Remove whitespace

claywar committed:
[lua] Fix items not dropping, add TODO for Yellow and Blue staggers

zach2good committed:
[abyssea] Start on NM: Briareus

claywar committed:
[core] Add CLuaStatusEffect::getIcon() binding

claywar committed:
[lua] Prototype searing ward tether, and initial 5 minutes of visitant on entry

claywar committed:
[lua] Prototype time remaining reporting for visitant status


Remove debug prints


Update comments and always reset last update on time change

claywar committed:
[core] Add support for Abyssea Maw unlocks

claywar committed:
[lua] Add enum for maw unlock bitfield, unlock after zone in, and refactor Warp NPC

claywar committed:
[lua] Move abyssea warps to global, add remaining Warp NPCs


Add missing onEventFinish parameter

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
[lua] Prototype Traverser Stone NPCs, minor bugfixes

claywar committed:
[lua] Rename functions for clarity, add additional comments

claywar committed:
[lua] Move Traverser NPC functions to abyssea global


Fix Syntax in Global


Add missing require

claywar committed:
[lua] Wire up remaining Traverser NPCs and add Heavens Tower NPCs

claywar committed:
[lua] Add Scars of Abyssea Quest


Misc Cleanup for Quest

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
[lua] Update visitant status to eject to the proper zone on timeout

claywar committed:
[lua] Convert Exiting text IDs to offsets, enable searing wards for A-L

claywar committed:
[lua] Enable searing ward tether for M-V

claywar committed:
[lua] Add Cruor Prospector items to items global

claywar committed:
[lua] Add remaining Cruor Prospectors, fix Forbidden Keys for visions areas

claywar committed:
[core] Add support for EFFECTFLAG_HIDE_TIMER


Add leading zeroes to effectflag enum for clarity

claywar committed:
[lua] Prototype Conflux Surveyor, add additional text offsets, and reset lights on effect loss

claywar committed:
[core/lua] Remove abyssea_maws core references and db column and replace with quest check

claywar committed:
[lua] Move Conflux Surveyor to abyssea global

claywar committed:
[core] Disable dedication bonuses inside Abyssea

claywar committed:
[core/lua] Move Abyssea Conflux masks to char_unlocks

claywar committed:
[lua] Minor pyxis cleanup, add missing text IDs


Fix time extension chests for initial setpoint

claywar commented:
Things are starting to get into a decently testable state.  There's still a ton to do, but would appreciate some eyes to provide feedback.

claywar committed:
[lua] Fix formatting for multiline messages by changing messageSpecial to messageName

claywar committed:
[lua] Move atma, pyxis, and moblights global to sub directory


[lua] Move abyssea atma global to sub directory


Remove errant remnant for pyxis global

claywar committed:
[lua] Break out conflux surveyor into sub-dir global

claywar committed:
[lua] Prototype Dominion Tactician - Abyssea Grauberg


Trailing whitespace

claywar committed:
[lua] Start blocking out Dominion Sergeant

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
[lua] Add Dominion Sergeant NPC scripts and bitmasks for available ops

claywar committed:
[core/lua] Eject if visitant lost while offline, grant GM flagged players automatic visitant


Additional debug code so that CI doesn't complain

claywar committed:
[lua] Wire up Dominion Op completion and cancellation (does not include seal reward)


Add additional text IDs for Grauberg

claywar committed:
[lua] Add quests scripts for Grauberg dominion ops

claywar committed:
[lua/sql] Add Visions of Abyssea main quests

claywar committed:
[lua] Add Pyxis NPC IDs

claywar committed:
[sql] Add additional NM levels and NPCs from captures

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

zach2good committed:
[abyssea] Clean up NM: Briareus

claywar committed:
Merge pull request #633 from zach2good/nm-briareus

[abyssea] NM: Briareus

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Fix additional mobskills after monstertpmoves changes


Fix mercurial_strike require

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

TeoTwawki commented on scripts/globals/abyssea.lua:
```diff
@@ -1,91 +1,494 @@
+
+    -- Konschtat Highlands
+	[16838751] = {['Atma'] = {                                         }, ['Normal'] = { xi.keyItem.FRAGRANT_TREANT_PETAL      }},
+	[16838855] = {['Atma'] = { xi.keyItem.ATMA_OF_THRASHING_TENDRILS   }, ['Normal'] = { xi.keyItem.FETID_RAFFLESIA_STALK      }},
```
can you use ki instead of keyItem to cut down the width a bit? (one's just alias for the other)

TeoTwawki commented on scripts/globals/abyssea.lua:
```diff
@@ -1,91 +1,494 @@
+xi.abyssea.mob =
+{
+    -- Attohwa Chasm
+	[17658287] = {['Atma'] = {                                          }, ['Normal'] = { xi.keyItem.SHRIVELED_HECTEYES_STALK   }},
```
going on record to say I despise the fact we have raw ID here instead of just handing it out at the entity it comes from (Amun in this particular row's case)

ShiyoKozuki commented on scripts/globals/abyssea/lights.lua:
```diff
@@ -0,0 +1,652 @@
+        ["Gasher"]                 = { azure = 8,   pearl = 5,   ruby = 8,   amber = 8  }, -- Crabs
+        ["Maritime_Peiste"]        = { azure = 16,  pearl = 5,   ruby = 8,   amber = 8  }, -- Peistes
+        ["Escarp_Murex"]           = { azure = 8,   pearl = 5,   ruby = 0,   amber = 8  }, -- Murex
+        ["Ephemeral_Murex"]        = { azure = 100, pearl = 5,   ruby = 16,  amber = 16 }, -- Murex
```
Shouldn't this be 128 azure?
https://ffxiclopedia.fandom.com/wiki/Ephemeral_Murex
https://www.bg-wiki.com/ffxi/Abyssea_Lights

TeoTwawki replied:
Nope. Retail does not consistently give you 128, nor is this 100 the amount you will actually get. The script uses 50 and 100 as "special" numbers to note a variance/range of results.

_I'll take #shitWikiSays for 100, Ghost of Alex Trebek_

TeoTwawki replied:
Additionally I should mention this is a systems work, clay is taking exactly what we had but making the code easier to deal with future forward - any preexisting error is likely to remain till after he's finished and be worked on separately

ShiyoKozuki replied:
Yeah number was UP to 128 iirc but I had no proof to back that up.

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
CI: Fix trailing whitespace after merge

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

Chronos77 committed:
Start atma fabricant and review npc for abyssea latheine and konschat

Chronos77 committed:
Remove logs

Chronos77 committed:
Fix whitespace

Chronos77 committed:
Fix synthax

Chronos77 committed:
Fix kukulkan required

Chronos77 committed:
Split sturdypyxis.lua for reduce complexity
Rewrite some code for sturdy pyxis
Split files for sturdy pyxis
Replace xi.keyItem to xi.ki
Clean some code for coffer blue and red, review drop rate, review drop by tier, add common drop for item
Add pop item coffer
Implement canOpenChest

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

Chronos77 committed:
Update name for npc

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Fix merge conflict issues in items.lua

Chronos77 committed:
Fix missing letter

claywar committed:
Merge pull request #987 from Chronos77/feature/abyssea-latheine-konschtat

Add some pop Abyssea-Konschtat and fix how to obtain KI and Atma

claywar committed:
Fix sol::nil reference for MacOS builds

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Update conflicts from November Shift

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

Chronos77 committed:
Add new line on the end of file

claywar committed:
Merge pull request #1023 from Chronos77/feature/abyssea-coffer

[Abyssea] Fix behavior for coffer and split files for sturdypyxis

claywar committed:
Merge remote-tracking branch 'origin/base' into update/abyssea

claywar committed:
Update Abyssea NPCs for December

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Update IDs strings to conform with tooling

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Update additional Abyssea Text ID comments


Update text IDs for January update


Additional ID comment updates


ID shifts for the remaining Abyssea Zones

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into feature/abyssea

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Delete old global to track changes

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Update IDs in Feature for tracking

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar committed:
Merge remote-tracking branch 'origin/base' into abyssea-update

claywar commented:
Closing this PR.  We've now reached parity with base after the last open item is merged.

zach2good commented:
_noice_

On Sat, Apr 16, 2022, 20:24 claywar ***@***.***> wrote:

> Closed #632 <https://github.com/LandSandBoat/server/pull/632>.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/LandSandBoat/server/pull/632#event-6443639008>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKTJIIKAPB2A5NWGMJTY63VFLZVNANCNFSM5CP6NMUA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
